src/dmd/declaration.d

```diff
@@ -362,6 +362,14 @@ extern (C++) abstract class Declaration : Dsymbol
                                     return false;
                         }
                     }
+                    if (auto ctor = isCtorDeclaration())
+                    {
+                        if (ctor.isCpCtor && ctor.generated)
+                        {
+                            .error(loc, "Generating an `inout` copy constructor for `struct %s` failed, therefore instances of it are uncopyable", parent.toPrettyChars());
+                            return true;
```

Comment:
This early return is good but the whole function should use it.

Reply:
I know, but I didn;t want to mix refactorings in this PR.